lib/ae_mdw/db/origin.ex

```diff
@@ -45,7 +45,7 @@ defmodule AeMdw.Db.Origin do
     and created through core hard-forks. These contracts will have a negative
     txi where the number is the index of the preloaded contracts.
   """
-  @spec tx_index(State.t(), creation_txi_locator()) :: {:ok, Txs.txi() | -1} | :not_found
+  @spec tx_index(State.t(), creation_txi_locator()) :: {:ok, integer()} | :not_found
```

Comment:
What would be the motivation behind if the Txs.txi() was used in so many places and the function returns a transaction index?

Reply:
`tx_index` returns a negative integer when the contract is one of the preset contracts: https://github.com/aeternity/ae_mdw/blob/1bedf79e4f95b0aeb2043a9614e128138c72fc29/lib/ae_mdw/db/origin.ex#L55

Reply:
Two alternatives to make it more clear from my perspective: 
a) if a negative index is accessible in the database (present somehow in `Model.tx`), then the `Txs.txi()` should include them.
b) if that's not the case, they should be returned in a tagged value instead of negative integer. I understand in this case it wouldn't be part of the scope of the PR but it's something for us to keep in mind when doing a refactoring.

Reply:
When the returned value of `tx_index` is for example `-8`, it means that the contract is the 8th preset contract. I agree though that this isn't really friendly, we should def do something about it. But the intention of this PR isn't a refactor, but to include more helpful typespecs of the current (unchanged) implementation

Reply:
Yes, totally fine to be fixed later.